src/NumericalAlgorithms/Interpolation/RegularGridInterpolant.hpp

```diff
@@ -0,0 +1,119 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Matrix.hpp"  // IWYU pragma: keep
+#include "DataStructures/Variables.hpp"
+#include "NumericalAlgorithms/LinearOperators/ApplyMatrices.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+
+// IWYU pragma: no_forward_declare Variables
+
+/// \cond
+template <size_t Dim>
+class Mesh;
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace intrp {
+
+/// \ingroup NumericalAlgorithmsGroup
+/// \brief Interpolate a Variables from a Mesh onto a regular grid of points.
+///
+/// The target points must lie on tensor-product grid that is aligned with the
+/// source Mesh. In any direction where the source and target points share an
+/// identical Mesh, (i.e., where the underlying 1-dimensional meshes share the
+/// same extent, basis, and quadrature), the code is optimized to avoid
+/// performing identity interpolations.
+/// Note, however, that in each dimension of the target grid, the points can
+/// be freely distributed; in particular, the grid points need not be the
+/// collocation points corresponding to a particular basis and quadrature.
+///
+/// \note When interpolating between two different resolutions on the same type
+/// of spectral grid (e.g., between two Legendre Gauss Lobatto grids with
+/// different numbers of points), it may be more efficient to use a projection
+/// instead of the interpolation that is done by this class. This class is
```

Comment:
The main reason to choose between projection and interpolation is that they give different answers for some parameters and you should pick the correct operation for your current task.  You shouldn't suggest that projection can be used as a drop-in replacement for interpolation in general.

Reply:
Agreed. Comment removed.